modules/810-ui/images/backend/cloudprovider/cluster_config.go

```diff
@@ -0,0 +1,62 @@
+package cloudprovider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/yaml"
+
+	"deckhouse.io/uibackend/util/retry"
+)
+
+type ProviderConfig struct {
+	Config        map[string]interface{}
+	DiscoveryData map[string]interface{}
+}
+
+func (c *ProviderConfig) Zones() []string {
+	zones, ok, _ := unstructured.NestedStringSlice(c.DiscoveryData, "zones")
+	if !ok {
+		return make([]string, 0)
+	}
+
+	return zones
+}
+
+func GetClusterConfig(ctx context.Context, client *kubernetes.Clientset) (*ProviderConfig, error) {
+	var secret *v1.Secret
+	err := retry.NewLoop("Get provider cluster config", 3, 3*time.Second).Run(func() error {
```

Comment:
Let's remove all 'retry' thing because for now we need only contract, but later we'll have informer subscription for discovery data